addon/components/lt-body.js

```diff
@@ -266,17 +267,31 @@ export default Component.extend({
       let isSelected = row.get('selected');
       let currIndex = rows.indexOf(row);
       let prevIndex = this._prevSelectedIndex === -1 ? currIndex : this._prevSelectedIndex;
+      let affectedSelections = Ember.A();
 
       this._currSelectedIndex = currIndex;
       this._prevSelectedIndex = prevIndex;
 
       if (canSelect) {
         if (e.shiftKey && multiSelect) {
-          rows.slice(Math.min(currIndex, prevIndex), Math.max(currIndex, prevIndex) + 1).forEach(r => r.set('selected', !isSelected));
+          rows.slice(Math.min(currIndex, prevIndex), Math.max(currIndex, prevIndex) + 1).forEach(r => {
+            if (!r.get('selected')) {
+                affectedSelections.pushObject(r);
+            }
+
+            r.set('selected', !isSelected);
+          });
           this._prevSelectedIndex = currIndex;
         } else if ((!multiSelectRequiresKeyboard || (e.ctrlKey || e.metaKey)) && multiSelect) {
           row.toggleProperty('selected');
+          affectedSelections.pushObject(row);
         } else {
+          if (!isSelected) {
+            affectedSelections.pushObject(row);
```

Comment:
> To avoid adding the target row to the affectedSelections array twice

Like I said, we should be adding ALL rows that have changed EXCLUDING the current row. So you can return something like this:

```js
affectedSelections.uniq().without(currentRow)
```

>  I think it might be better to send them as a separate array parameter, e.g. affectedRowExpansions or something like that

What if we send a `changes` hash?

```js
{
  selected: [ ],
  expanded: [ ]
}
```